managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1980,7 +1982,7 @@ void internalMarkDelete(final MarkDeleteEntry mdEntry) {
         }
 
         PositionImpl inProgressLatest = INPROGRESS_MARKDELETE_PERSIST_POSITION_UPDATER.updateAndGet(this, current -> {
-            if (current != null && current.compareTo(mdEntry.newPosition) > 0) {
+            if (current != null && current.compareTo(mdEntry.newPosition) >= 0) {
```

Comment:
Changing the 1993 line with the equals method is better because it is weird to use object address to determine.

Reply:
OK， I will fix